tests/test_trace_exporter.py

```diff
@@ -114,7 +112,66 @@ def test_export(self, mock_tracer):
             )
 
-            # test parent hierarchy
-            self.assertIsNotNone(result_spans[0].parent_id)
-            self.assertEqual(result_spans[0].parent_id, result_spans[1].context.span_id)
-            self.assertIsNone(result_spans[1].parent_id)
+        # test parent hierarchy
+        self.assertIsNotNone(result_spans[0].parent_id)
+        self.assertEqual(result_spans[0].parent_id, result_spans[1].context.span_id)
+        self.assertIsNone(result_spans[1].parent_id)
+
+    @patch("lightstep.Tracer")
+    def test_span_attributes(self, mock_tracer):
+        """ ensure span attributes are passed as tags """
+        otel_span = trace.Span(name=__name__, context=self._span_context)
+        otel_span.set_attribute("key_bool", False)
+        otel_span.set_attribute("key_string", "hello_world")
+        otel_span.set_attribute("key_float", 111.22)
+        result_spans = self._process_spans([otel_span])
+        self.assertEqual(len(result_spans), 1)
+        self.assertEqual(len(result_spans[0].tags), 3)
+        self.assertFalse(result_spans[0].tags.get("key_bool"))
+        self.assertEqual(result_spans[0].tags.get("key_string"), "hello_world")
+        self.assertEqual(result_spans[0].tags.get("key_float"), 111.22)
+
+    @patch("lightstep.Tracer")
+    def test_events(self, mock_tracer):
+        """ test events are translated into logs """
+        base_time = 683647322 * 10 ** 9  # in ns
+        event_attributes = {
+            "annotation_bool": True,
+            "annotation_string": "annotation_test",
+            "key_float": 0.3,
+        }
+
+        event_timestamp = base_time + 50 * 10 ** 6
+        event = trace.Event(
+            name="event0", timestamp=event_timestamp, attributes=event_attributes,
+        )
+        otel_span = trace.Span(
+            name=__name__, context=self._span_context, events=(event,),
+        )
+        result_spans = self._process_spans([otel_span])
+        self.assertEqual(len(result_spans), 1)
+        self.assertEqual(len(result_spans[0].logs), 1)
+        log = result_spans[0].logs[0]
+        # timestamp is in seconds, event_timestamp in ns
+        self.assertEqual(log.timestamp, (event_timestamp / 1000000000))
+        self.assertEqual(log.key_values, event_attributes)
+
+    @patch("lightstep.Tracer")
+    def test_links(self, mock_tracer):
+        """ test links are translated into references """
+        # TODO
+
+    #     link_attributes = {"key_bool": True}
+
+    #     link = trace_api.Link(context=self._span_context, attributes=link_attributes)
+    #     otel_span = trace.Span(
+    #         name=__name__, context=self._span_context, links=(link,),
+    #     )
+    #     result_spans = self._process_spans([otel_span])
+    #     self.assertEqual(len(result_spans), 1)
+    #     self.assertEqual(len(result_spans[0].references), 1)
+
+    @patch("lightstep.Tracer")
+    def test_resources(self, mock_tracer):
+        """ test resources """
+        # TODO
```

Comment:
```suggestion
```

Empty tests are still collected, executed and reported.

Reply:
i think that's fine, I will be filling those in before the next release.